src/test/groovy/graphql/GraphQLTest.groovy

```diff
@@ -45,17 +47,19 @@ class GraphQLTest extends Specification {
                 .field(
                 newFieldDefinition()
                         .name("id")
+                        .dataFetcher(new MapDataFetcher("id"))
```

Comment:
> I get that an object has fields that can be "fetched".

Well, an object only has fields in the Java sense of the word "field" if it has public fields that can be accessed like in [`FieldDataFetcher`](https://github.com/dminkovsky/graphql-java/blob/eb75d57f0bbcaed455812218fd81beda028a8379/src/main/java/graphql/schema/FieldDataFetcher.java#L26), using reflection.

> Those fields could potentially be fetched as properties too 

Yes, if they can be fetched as properties too (because the object has getter methods that return the field values), you can alternatively use `PropertyDataFetcher`. 

What I am trying to do here is to avoid having something called `PropertyDataFetcher` that actually does 3 different things.